src/core/matchers/matchersUtil.js

```diff
@@ -223,10 +223,14 @@ getJasmineRequireObj().matchersUtil = function(j$) {
       }
 
       var extraKeys = [];
```

Comment:
Do we need to initialize this when allKeys is empty? We can move lines 226-228 before line 224.